blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxLocatorIO.java

```diff
@@ -68,11 +71,21 @@ public void insertLocator(Locator locator) throws IOException {
     @Override
     public Collection<Locator> getLocators(long shard) throws IOException {
         Timer.Context ctx = Instrumentation.getReadTimerContext(CassandraModel.CF_METRICS_LOCATOR_NAME);
+        final Collection<Locator> locators = new ArrayList<Locator>();
         try {
             RowQuery<Long, Locator> query = AstyanaxIO.getKeyspace()
                     .prepareQuery(CassandraModel.CF_METRICS_LOCATOR)
                     .getKey(shard);
-            return query.execute().getResult().getColumnNames();
+
+            ColumnList<Locator> columns = query.execute().getResult();
+            for (Column<Locator> column: columns) {
+                locators.add(column.getName().withLastUpdatedTimestamp(column.getTimestamp() / 1000));
```

Comment:
Cassandra's timestamp is in micro seconds. I am converting it to milli seconds as all our timestamps are in milli seconds.